fv3gfs/util/communicator.py

```diff
@@ -443,13 +474,16 @@ def _get_scatter_recv_quantity(
         return recv_quantity
 
     def halo_update(self, quantity: Quantity, n_points: int):
+        self.halo_update_aggregate([quantity], n_points)
+
+    def halo_update_aggregate(self, quantities: List[Quantity], n_points: int):
```

Comment:
I would suggest refactoring this to `halo_update(self, *quantities, n_points: int)` to reduce the number of methods to remember and test. It makes n_points a keyword-only argument but that's fine, just be aware it's a backwards-incompatible change. You could keep backwards compatibility with a warning by checking if the last entry in `quantities` is an integer.

You can similarly refactor the other APIs.

Reply:
I refactored those to take a `Union[Quantity, List[Quantity]]` and check on an `isinstance()` to make a `List[Quantity]`. That allows for API backwards compatibility at the cost of a `isinstance`. That said, the next chunk of work will have an impact on the API (moving to objects)